ambry-store/src/test/java/com.github.ambry.store/IndexValueUtils.java

```diff
@@ -0,0 +1,148 @@
+
+  /**
+   * Constructs IndexValue based on the size and {@link Offset} for {@link PersistentIndex#VERSION_0}
+   * @param size the size of the blob that this index value refers to
+   * @param offset the {@link Offset} in the {@link Log} where the blob that this index value refers to resides
+   * @return the {@link IndexValue} thus constructed
+   */
+  private static IndexValue getIndexValue(long size, Offset offset) {
+    return getIndexValue(size, offset, (byte) 0, Utils.Infinite_Time, offset.getOffset());
+  }
+
+  /**
+   * Constructs IndexValue based on the args passed in version {@link PersistentIndex#VERSION_0}
+   * @param size the size of the blob that this index value refers to
+   * @param offset the {@link Offset} in the {@link Log} where the blob that this index value refers to resides
+   * @param expiresAtMs the expiration time in ms at which the blob expires
+   * @param originalMessageOffset the original message offset where the Put record pertaining to a delete record exists
+   *                              in the same log segment. Set to -1 otherwise.
+   * @return the {@link IndexValue} thus constructed
+   */
+  private static IndexValue getIndexValue(long size, Offset offset, byte flags, long expiresAtMs,
+      long originalMessageOffset) {
+    ByteBuffer value = ByteBuffer.allocate(IndexValue.INDEX_VALUE_SIZE_IN_BYTES_V0);
+    value.putLong(size);
+    value.putLong(offset.getOffset());
+    value.put(flags);
+    value.putLong(expiresAtMs);
+    value.putLong(originalMessageOffset);
+    value.position(0);
+    return new IndexValue(offset.getName(), value, PersistentIndex.VERSION_0);
+  }
+
+  /**
+   * Constructs IndexValue based on the args passed in the version {@link PersistentIndex#VERSION_0}
+   * @param size the size of the blob that this index value refers to
+   * @param offset the {@link Offset} in the {@link Log} where the blob that this index value refers to resides
+   * @param flags the flags to be set in the {@link IndexValue}
+   * @param expiresAtMs the expiration time in ms at which the blob expires
+   * @return the {@link IndexValue} thus constructed
+   */
+  private static IndexValue getIndexValue(long size, Offset offset, byte flags, long expiresAtMs) {
+    return getIndexValue(size, offset, flags, expiresAtMs, offset.getOffset());
+  }
+
+  /**
+   * Constructs IndexValue based on the args passed in the version {@link PersistentIndex#VERSION_0}
+   * @param size the size of the blob that this index value refers to
+   * @param offset the {@link Offset} in the {@link Log} where the blob that this index value refers to resides
+   * @param expiresAtMs the expiration time in ms at which the blob expires
+   * @return the {@link IndexValue} thus constructed
+   */
+  private static IndexValue getIndexValue(long size, Offset offset, long expiresAtMs) {
+    return getIndexValue(size, offset, (byte) 0, expiresAtMs, offset.getOffset());
+  }
+}
```

Comment:
there are too many functions in this class. These are tests so I think having one function is more than enough. If callers don't need certain variables, they can pass in default values (which can be defined as static variables for easy use).